DOC add BaggingClassifier Examples

JesperDramsch opened the pull request:
Toward #3846 
this PR adds an example to `ensemble.BaggingClassifier`

ping @adrinjalali

JesperDramsch committed:
Bagging Classifier Examples in Doc

JesperDramsch commented:
I honestly don't understand why the classifier output for the Doctest fails in the azure pipeline. I could use some clarification on those.

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,39 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    ...                            random_state=0, shuffle=False)
+    >>> clf = BaggingClassifier(n_estimators=100, random_state=0)
+    >>> clf.fit(X, y)  # doctest: +NORMALIZE_WHITESPACE
+    BaggingClassifier(base_estimator=None, bootstrap=True,
```
It's odd that some fail and some don't, but for the example, I'd just put the fit right after construction of the class and put it in `clf`, and as a result you don't get the pprint of the object, and that should solve the issue anyway.

JesperDramsch committed:
Omit pprint 

Azure Pipeline partially breaks on the pprint

Left the response in as a comment, I think they make the use a bit clearer

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,37 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    >>> #                   max_samples=1.0, n_estimators=100,
+    >>> #                   n_jobs=None, oob_score=False,
+    >>> #                   random_state=0, verbose=0,
+    >>> #                   warm_start=False)
```
I know some of the examples have these, but we don't really need them, and they cause more trouble than they help. You can just remove them.

JesperDramsch replied:
Ok

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,37 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    >>> #                   n_jobs=None, oob_score=False,
+    >>> #                   random_state=0, verbose=0,
+    >>> #                   warm_start=False)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
```
```suggestion
    >>> clf.predict([[0, 0, 0, 0]])
```

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,37 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    >>> #                   max_features=1.0, max_samples=1.0,
+    >>> #                   n_estimators=100, n_jobs=None,
+    >>> #                   oob_score=False, random_state=0,
+    >>> #                   verbose=0, warm_start=False)
```
remove these too

JesperDramsch replied:
Ok

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,37 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    >>> #                   n_estimators=100, n_jobs=None,
+    >>> #                   oob_score=False, random_state=0,
+    >>> #                   verbose=0, warm_start=False)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
```
```suggestion
    >>> clf.predict([[0, 0, 0, 0]])
```

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,37 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    >>> #                   warm_start=False)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
+    [1]
+    >>> clf = BaggingClassifier(base_estimator=DummyClassifier(),
```
I would use something smarter than _Dummy_ ;) Since these code snippets get copy/pasted, using a more sensible option wouldn't be a bad idea.

JesperDramsch replied:
Done

adrinjalali reviewed (commented):
Thanks :)

JesperDramsch committed:
Apply suggestions from code review

Co-Authored-By: Adrin Jalali <adrin.jalali@gmail.com>

JesperDramsch committed:
Change Classifier and remove comment

JesperDramsch commented:
Thank you for your help @adrinjalali

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,23 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
+    ...                            n_informative=2, n_redundant=0,
+    ...                            random_state=0, shuffle=False)
+    >>> clf = BaggingClassifier(n_estimators=100, random_state=0).fit(X, y)
+    >>> print(clf.predict([[0, 0, 0, 0]]))
```
You don't need to `print`, you would see the output in the REPL anyway.

JesperDramsch replied:
I know, I just followed the suggestion from your last review. Should I revert this?

adrinjalali replied:
I thought my suggestion was to remove it :P

![suggestion](https://user-images.githubusercontent.com/1663864/66210836-080f3d00-e6bb-11e9-91e4-48cccff2fbbb.png)

JesperDramsch replied:
Ha. I got confused, sorry about that.

adrinjalali reviewed (approved):
Nit, otherwise LGTM. Thanks @JesperDramsch

JesperDramsch committed:
Remove print

adrinjalali commented:
Tests failing because when you remove the print, the output becomes `array([1])` instead of `[1]`. You need to fix the outputs as well :)

JesperDramsch committed:
Change output

adrinjalali commented on sklearn/ensemble/bagging.py:
```diff
@@ -541,6 +541,23 @@ class BaggingClassifier(ClassifierMixin, BaseBagging):
         `oob_decision_function_` might contain NaN. This attribute exists
         only when ``oob_score`` is True.
 
+
```
extra line

adrinjalali reviewed (approved):
otherwise LGTM, thanks @JesperDramsch

rth committed:
Remove extra line

rth reviewed (approved):
Thanks @JesperDramsch !
